usr/lib/common/mech_openssl.c

```diff
@@ -4898,6 +4898,325 @@ CK_RV openssl_specific_hmac_final(SIGN_VERIFY_CONTEXT *ctx, CK_BYTE *signature,
     return rv;
 }
 
+CK_RV calc_rsa_crt_from_me(CK_ATTRIBUTE *modulus, CK_ATTRIBUTE *pub_exp,
+                           CK_ATTRIBUTE *priv_exp, CK_ATTRIBUTE **prime1,
+                           CK_ATTRIBUTE **prime2, CK_ATTRIBUTE **exponent1,
+                           CK_ATTRIBUTE **exponent2, CK_ATTRIBUTE **coef)
+{
```

Comment:
Just to double check: you already checked that either all CRT components are given, or none of them. Otherwise the calculation could either rely on already given components or check if they are identical to the calculated ones.

Reply:
Correct, a key can only be created with either all or none of them.